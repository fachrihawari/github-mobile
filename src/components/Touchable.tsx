import React, { ReactNode, ReactType } from "react";
import {
  TouchableOpacity,
  TouchableNativeFeedback,
  Platform
} from "react-native";

interface ITouchableProps {
  onPress: Function;
  children: ReactNode;
}

function Touchable({ children, onPress }: ITouchableProps) {
  const Component: ReactType =
    Platform.OS === "ios" ? TouchableOpacity : TouchableNativeFeedback;

  return <Component onPress={onPress}>{children}</Component>;
}

export default Touchable;
