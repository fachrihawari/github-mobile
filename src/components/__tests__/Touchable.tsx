import "react-native";
import { Text } from "react-native";
import React from "react";
import render from "react-test-renderer";

import Touchable from "../Touchable";

it("render default Touchable", () => {
  const handlePress = jest.fn();
  const tree = render
    .create(
      <Touchable onPress={handlePress}>
        <Text>Awesome Touchable</Text>
      </Touchable>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
