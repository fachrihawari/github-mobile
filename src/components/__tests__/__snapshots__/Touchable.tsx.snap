// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render default Touchable 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text>
    Awesome Touchable
  </Text>
</View>
`;
